lms/djangoapps/course_home_api/course_metadata/v1/views.py

```diff
@@ -104,7 +98,8 @@ def get(self, request, *args, **kwargs):
             'is_self_paced': getattr(course, 'self_paced', False),
             'is_enrolled': user_is_enrolled,
             'can_load_courseware': can_load_courseware,
-            'celebrations': celebrations,
+            'celebrations': courseware_meta.celebrations,
```

Comment:
it would be more DRY/simpler to reuse celebrations from the courseware_api code
the one downside is the dictionary of celebrations would include a first_section_celebration field that wouldn't be used on course home - but also if the page just doesn't use that field it might not be the biggest deal
the other two fields in the dict right now are used in both courseware/course home
thoguhts?

Reply:
This is an interesting case and I could see it both ways.
- These two djangoapps are closely related, so sharing code isn't _the worst_
- But the `CoursewareMeta` class feels like an implementation detail fueling their view, and less an API meant to be consumed.
- Being DRY is great, but in this case, we we're already calling out to a shared utility function for the business logic. I'm not super worried about duplicating a dictionary construction.
- But, as that dictionary grows, it starts getting gross to duplicate the construction. And you've grown the dictionary in this PR. Plus it's probably good to ensure that both rest APIs look the same, so we can share some code on the frontend side.
- But even then, I don't know if `CoursewareMeta` is the best vessel for sharing this code, rather than like, sharing some more-obviously-scoped utility methods. That whole class has a whole bunch of stuff, and sharing-creep is a thing. Like here.

Anyway. I might slightly prefer breaking this shared code out to a utility method rather than the class (like how both sides' `verified_mode` is powered by `serialize_upgrade_info`). But class is also fine. :shrug:

Reply:
I feel like long term you could have something like CoursewareMeta but specifically intended for sharing the code that might allow fields to be easily moved back and forth between the different APIs where the APIs just say which fields they want rather than defining each field individually

but sure i can use a utility in this case

Reply:
Does it feel cleaner do a mixin, like for verified mode? A cross between a class and a utility function?  :P

Reply:
yeah i think a mixin would certainly be ideal for what I was describing 
when I said like CoursewareMeta I just meant having one place where a lot of the data is defined - so you'd have a MetadataMixin and then different metadata views could pull in the fields they need from there